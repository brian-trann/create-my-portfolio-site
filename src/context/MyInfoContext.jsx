import React from 'react';
const MyInfoContext = React.createContext();
export default MyInfoContext;
