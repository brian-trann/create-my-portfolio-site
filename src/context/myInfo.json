{
    "navigation": {
        "name": "John Doe",
        "menu": [
            {
                "text": "Home",
                "url": "/",
                "target":""
            },
            {
                "text": "About",
                "url": "/about",
                "target":""
            },
            {
                "text": "Github",
                "url": "https://github.com/",
                "target":"_blank"
            },
            {
                "text": "LinkedIn",
                "url": "https://www.linkedin.com//",
                "target":"_blank"
            }
        ]
    },
    "main": {
        "greeting": "Hi, I'm John, a Software Developer from California",
        "blurb":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
        "profileImage": { 
            "url":"",
            "altText":"profile image"
        },
        "image": {
            "url": "/",
            "altText": "temp"
        }
    },
    "work":{
        "categoryName":"Selected Work",
        "projects":[
            {
                "projectName": "project0",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project0 thumbnail"
                }
            },
            {
                "projectName": "project1",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project1 thumbnail"
                }
            },
            {
                "projectName": "project2",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project2 thumbnail"
                }
            }
        ],
        "buttonNames":{
            "primary":"Github",
            "secondary":"Site"
        }
    },
    "about": {
        "categoryName":"About",
        "statement": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
        "skills":{
            "categoryName":"Skills",
            "items":[
                {
                    "type":"Languages",
                    "items":["JavaScript", "HTML", "CSS","Python","SQL"]
                },
                {
                    "type":"Libraries and Frameworks",
                    "items":["React", "Redux", "Express","Flask"]
                },
                {
                    "type":"Other",
                    "items":["other0", "other1"]
                }
            ]
        },
        "workExperience": {
            "categoryName":"Work Experience",
            "work": [
                {
                    "company":"company0",
                    "position":"position0",
                    "date":"Date 2020 - Date 2021",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                },
                {
                    "company":"company2",
                    "position":"position1",
                    "date":"Date 2019 - Date 2020",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                }
            ]
        },
        "education":{
            "categoryName":"Education",
            "items": [
                {
                    "name":"education0",
                    "curriculum":"curriculum0",
                    "date":"Date 2020 - Date 2021",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                }
            ]
        },
        "certifications":{
            "categoryName":"Certifications",
            "items":["certification0", "certification1"]
        }
    },
    "other":{
        "images":[
            {"image":{
                "url":"https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                "altText":"",
                "description":""
                }
            }
        ]
    }
}