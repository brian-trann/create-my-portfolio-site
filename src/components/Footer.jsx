import React, { useContext } from 'react';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import MyInfoContext from '../context/MyInfoContext';

const useStyles = makeStyles((theme) => ({
	footer : {
		padding : theme.spacing(8)
	}
}));
const Footer = () => {
	const { navigation: { name } } = useContext(MyInfoContext);
	const classes = useStyles();
	return (
		<footer className={classes.footer}>
			<Typography variant='subtitle1' align='center' color='textSecondary' component='p'>
				{name} {new Date().getFullYear()}
			</Typography>
		</footer>
	);
};
export default Footer;
