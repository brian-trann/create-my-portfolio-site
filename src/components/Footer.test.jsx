import React from 'react';
import { render } from '@testing-library/react';
import Footer from './Footer';
import myInfo from '../context/myInfo.json';
import MyInfoContext from '../context/MyInfoContext';

// smoke test
it('renders without crashing', () => {
	render(
		<MyInfoContext.Provider value={myInfo}>
			<Footer />
		</MyInfoContext.Provider>
	);
});

// snapshot test
it('matches snapshot', () => {
	const { asFragment } = render(
		<MyInfoContext.Provider value={myInfo}>
			<Footer />
		</MyInfoContext.Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
