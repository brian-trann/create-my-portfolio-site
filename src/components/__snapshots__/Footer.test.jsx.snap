// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="makeStyles-footer-2"
  >
    <p
      class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
    >
      John Doe 2021
    </p>
  </footer>
</DocumentFragment>
`;
